### PR TITLE
Optimized TwoColorAreaSeries performance and added two polygon mode

### PR DESCRIPTION
**Pull request by olegtarasov**
As discussed in #819, I added two polygons mode to TwoColorAreaSeries.

### CODE REVIEW

**Comment by olegtarasov** [comment]
So how about a merge on this PR? :)

**Comment by objorke** [comment]
Sorry I have not had much time available lately. Can anyone else have a look at this PR? It is quite a lot of changes! I keep it in my backlog!

**Comment by objorke** [comment]
Sorry for the delay. I have reviewed this code now, and it looks ok. I am a bit worried about the growing complexity of the AreaSeries, but it seems to be well organized code with small and well composed methods. Thanks for contributing! And for including an example!

File: Source/OxyPlot/Series/AreaSeries.cs [hunk 1]
```diff
@@ -75,7 +75,7 @@ public AreaSeries()
         /// Gets the actual color of the line for the second data set.
         /// </summary>
         /// <value>The actual color.</value>
-        public OxyColor ActualColor2
+        public virtual OxyColor ActualColor2
```
Comment (objorke) [inline Source/OxyPlot/Series/AreaSeries.cs]: I find it a bit strange that only the ActualColor2 property is virtual (why isn't ActualColor and ActualFill virtual...) But I see it is useful for the TwoColorAreaSeries.

File: Source/OxyPlot/Series/AreaSeries.cs [hunk 2]
```diff
@@ -238,11 +243,27 @@ public override void Render(IRenderContext rc)
             var clippingRect = this.GetClippingRect();
             rc.SetClip(clippingRect);
 
-            var chunksOfPoints = new List<List<ScreenPoint>>();
-            var chunksOfPoints2 = new List<List<ScreenPoint>>();
+            var areaContext = new AreaRenderContext
```
Comment (objorke) [inline Source/OxyPlot/Series/AreaSeries.cs]: this is a new pattern, but I think this is something that can be used in more places. Maybe more of the rendering code could use composition over inheritance?

File: Source/OxyPlot/Series/AreaSeries.cs [hunk 3]
```diff
@@ -141,6 +141,11 @@ protected List<DataPoint> ActualPoints2
         protected int WindowStartIndex2 { get; set; }
 
         /// <summary>
+        /// Gets a value indicating whether Points2 collection was defined by user.
+        /// </summary>
+        protected bool IsPoints2Defined { get; private set; }
```
Comment (objorke) [inline Source/OxyPlot/Series/AreaSeries.cs]: OK

File: Source/OxyPlot/Series/TwoColorAreaSeries.cs [hunk 4]
```diff
@@ -15,58 +15,103 @@ namespace OxyPlot.Series
     /// <summary>
     /// Represents a two-color area series.
     /// </summary>
-    public class TwoColorAreaSeries : TwoColorLineSeries
+    public class TwoColorAreaSeries : AreaSeries
```
Comment (objorke) [inline Source/OxyPlot/Series/TwoColorAreaSeries.cs]: Good

**Pull request merged**